contracts/strategies/StrategyCurveGUSDVoterProxy.sol

```diff
@@ -0,0 +1,223 @@
+
+    address public constant curve = address(0x0aE274c98c0415C0651AF8cF52b010136E4a0082);
+    address public constant gauge = address(0xC5cfaDA84E902aD92DD40194f0883ad49639b023);
+    address public constant voter = address(0xF147b8125d2ef93FB6965Db97D6746952a133934);
+
+    address public constant dai = address(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+    address public constant weth = address(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2); // used for crv <> weth <> dai route
+
+    uint256 public keepCRV = 1000;
+    uint256 public treasuryFee = 1000;
+    uint256 public strategistReward = 1000;
+    uint256 public withdrawalFee = 0;
+    uint256 public constant FEE_DENOMINATOR = 10000;
+
+    address public proxy;
+    address public dex;
+
+    address public governance;
+    address public controller;
+    address public strategist;
+    address public keeper;
+
+    uint256 public earned; // lifetime strategy earnings denominated in `want` token
+
+    event Harvested(uint256 wantEarned, uint256 lifetimeEarned);
+
+    constructor(address _controller) public {
+        governance = msg.sender;
+        strategist = msg.sender;
+        keeper = msg.sender;
+        controller = _controller;
+        // standardize constructor
+        proxy = address(0xC17ADf949f524213a540609c386035D7D685B16F);
+        dex = address(0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F);
+    }
+
+    function getName() external pure returns (string memory) {
+        return "StrategyCurveGUSDVoterProxy";
+    }
+
+    function setStrategist(address _strategist) external {
+        require(msg.sender == strategist || msg.sender == governance, "!authorized");
+        strategist = _strategist;
+    }
+
+    function setKeeper(address _keeper) external {
+        require(msg.sender == strategist || msg.sender == governance, "!authorized");
+        keeper = _keeper;
+    }
+
+    function setKeepCRV(uint256 _keepCRV) external {
+        require(msg.sender == governance, "!governance");
+        keepCRV = _keepCRV;
+    }
+
+    function setWithdrawalFee(uint256 _withdrawalFee) external {
+        require(msg.sender == governance, "!governance");
+        withdrawalFee = _withdrawalFee;
+    }
+
+    function setTreasuryFee(uint256 _treasuryFee) external {
+        require(msg.sender == governance, "!governance");
+        treasuryFee = _treasuryFee;
+    }
+
+    function setStrategistReward(uint256 _strategistReward) external {
+        require(msg.sender == governance, "!governance");
+        strategistReward = _strategistReward;
+    }
+
+    function setProxy(address _proxy) external {
+        require(msg.sender == governance, "!governance");
+        proxy = _proxy;
+    }
+
+    function switchDex(bool isUniswap) external {
+        require(msg.sender == strategist || msg.sender == governance, "!authorized");
+        // save gas and storage
+        if (isUniswap) {
+            dex = address(0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D);
```

Comment:
Could these be constants instead?

Reply:
don't we want to save some gas and storage?

Reply:
but to make it more readable, sure could make it. @lbertenasco also comments on this. Voting pass.

Reply:
Using a constant shouldn't waste any additional gas, it's just a label for this literal value, making it a bit easier to read.